client/src/hooks/generateZip.jsx

```diff
@@ -0,0 +1,21 @@
+import JSZip from  'jszip';
+import axios from "axios"
+import { saveAs } from 'file-saver';
+
+
+const generateZip =  async (/**@type Map<number, boolean>*/ m) => {
```

Comment:
Let's pass in the parameters of fileNames with type of Array<strings>, so we don't have to hard code the fileNames